core/ast.h

```diff
@@ -60,6 +60,101 @@ enum ASTType {
     AST_VAR
 };
 
+static inline std::string ASTTypeToString(ASTType type) {
+    switch(type) {
+        case AST_APPLY:
+        return "AST_APPLY";
+        break;
+        case AST_ARRAY:
+        return "AST_ARRAY";
+        break;
+        case AST_ARRAY_COMPREHENSION:
+        return "AST_ARRAY_COMPREHENSION";
+        break;
+        case AST_ARRAY_COMPREHENSION_SIMPLE:
+        return "AST_ARRAY_COMPREHENSION_SIMPLE";
+        break;
+        case AST_ASSERT:
+        return "AST_ASSERT";
+        break;
+        case AST_BINARY:
+        return "AST_BINARY";
+        break;
+        case AST_BUILTIN_FUNCTION:
+        return "AST_BUILTIN_FUNCTION";
+        break;
+        case AST_CONDITIONAL:
+        return "AST_CONDITIONAL";
+        break;
+        case AST_DESUGARED_OBJECT:
+        return "AST_DESUGARED_OBJECT";
+        break;
+        case AST_DOLLAR:
+        return "AST_DOLLAR";
+        break;
+        case AST_ERROR:
+        return "AST_ERROR";
+        break;
+        case AST_FUNCTION:
+        return "AST_FUNCTION";
+        break;
+        case AST_IMPORT:
+        return "AST_IMPORT";
+        break;
+        case AST_IMPORTSTR:
+        return "AST_IMPORTSTR";
+        break;
+        case AST_INDEX:
+        return "AST_INDEX";
+        break;
+        case AST_IN_SUPER:
+        return "AST_IN_SUPER";
+        break;
+        case AST_LITERAL_BOOLEAN:
+        return "AST_LITERAL_BOOLEAN";
+        break;
+        case AST_LITERAL_NULL:
+        return "AST_LITERAL_NULL";
+        break;
+        case AST_LITERAL_NUMBER:
+        return "AST_LITERAL_NUMBER";
+        break;
+        case AST_LITERAL_STRING:
+        return "AST_LITERAL_STRING";
+        break;
+        case AST_LOCAL:
+        return "AST_LOCAL";
+        break;
+        case AST_OBJECT:
+        return "AST_OBJECT";
+        break;
+        case AST_OBJECT_COMPREHENSION:
+        return "AST_OBJECT_COMPREHENSION";
+        break;
+        case AST_OBJECT_COMPREHENSION_SIMPLE:
+        return "AST_OBJECT_COMPREHENSION_SIMPLE";
+        break;
+        case AST_PARENS:
+        return "AST_PARENS";
+        break;
+        case AST_SELF:
+        return "AST_SELF";
+        break;
+        case AST_SUPER_INDEX:
+        return "AST_SUPER_INDEX";
+        break;
+        case AST_UNARY:
+        return "AST_UNARY";
+        break;
+        case AST_VAR:
+        return "AST_VAR";
+        break;
+        default:
+        assert(false && "Unknown AST type");
```

Comment:
It is better because the asserts can be disabled, and most people expect that to not change the behavior of the code.